test/plugins/metrics/node_config_test.go

```diff
@@ -0,0 +1,275 @@
+package metrics_test
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "github.com/rancher/opni/pkg/apis/core/v1"
+	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/test"
+	"github.com/rancher/opni/pkg/test/testutil"
+	"github.com/rancher/opni/plugins/metrics/pkg/apis/node"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+var _ = Describe("Node Config", Ordered, Label("integration"), func() {
+	var mgmtClient managementv1.ManagementClient
+	var nodeClient node.NodeConfigurationClient
+	BeforeAll(func() {
+		env := &test.Environment{
+			TestBin: "../../../testbin/bin",
+		}
+		Expect(env.Start(test.WithEnableCortexClusterDriver(true))).To(Succeed())
+
+		mgmtClient = env.NewManagementClient()
+
+		token, err := mgmtClient.CreateBootstrapToken(context.Background(), &managementv1.CreateBootstrapTokenRequest{
+			Ttl: durationpb.New(time.Hour),
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		certs, err := mgmtClient.CertsInfo(context.Background(), &emptypb.Empty{})
+		Expect(err).NotTo(HaveOccurred())
+
+		fp := certs.Chain[len(certs.Chain)-1].Fingerprint
+
+		_, errC := env.StartAgent("agent1", token, []string{fp}, test.WithAgentVersion("v2"), test.WithLocalAgent())
+		Eventually(errC).Should(Receive(BeNil()))
+
+		_, errC = env.StartAgent("agent2", token, []string{fp}, test.WithAgentVersion("v2"))
+		Eventually(errC).Should(Receive(BeNil()))
+
+		nodeClient = node.NewNodeConfigurationClient(env.ManagementClientConn())
+		DeferCleanup(env.Stop)
+	})
+
+	var getConfig = func(agentId string) (*node.MetricsCapabilitySpec, bool, error) {
+		var trailer metadata.MD
+		spec, err := nodeClient.GetNodeConfiguration(context.Background(), &v1.Reference{Id: agentId}, grpc.Trailer(&trailer))
+		if err != nil {
+			return nil, false, err
+		}
+		var isDefault bool
+		if len(trailer["is-default-config"]) > 0 {
+			isDefault = trailer["is-default-config"][0] == "true"
+		}
```

Comment:
might be worth putting this in a package since this default check is also used in the CLI